### PR TITLE
Add KVM integration tests

### PR DESCRIPTION
**Pull request by r2d4**
There still seem to be a few flakes.  I will fix those in additional commits.  This is non blocking

### CODE REVIEW

File: hack/jenkins/osx_integration_tests_xhyve.sh [hunk 1]
```diff
@@ -29,13 +29,18 @@ chmod +x out/e2e-darwin-amd64
 chmod +x out/minikube-darwin-amd64
 cp -r out/testdata ./
 
-
-./out/minikube-darwin-amd64 delete || true
+MINIKUBE_WANTREPORTERRORPROMPT=False \
+	./out/minikube-linux-amd64 delete || true
```
Comment (dlorenc) [inline hack/jenkins/osx_integration_tests_xhyve.sh]: darwin?

Comment (r2d4) [inline hack/jenkins/osx_integration_tests_xhyve.sh]: fixed

**Comment by r2d4** [comment]
@minikube-bot retest this please

**Comment by r2d4** [comment]
PTAL @dlorenc @aaron-prindle 

Additionally: should we add/change linux integration tests for systemd iso?

**Comment by dlorenc** [comment]
I'd like a hyper-v run and a KVM run at least once, but we don't need them in CI yet. Hopefully we'll fully switch over soon.

**Comment by r2d4** [comment]
We might as well add one for the linux slave since we are running 3 tests on the mac mini but only 2 on linux.

I'm also open to installing some of the binaries we need on the master to offload cross-building.  I'e already installed go 1.7.3 on the linux slave but what else do we need?

- go
- gsutil
- godep for localkube upgrades

**Comment by dlorenc** [comment]
Running builds on the slave shouldn't be necessary, we can just scale up the current master if we want. Are they a bottleneck anywhere?

**Comment by r2d4** [comment]
No, but `Execute concurrent builds` is unchecked on the `build cross` job so AFAIK they don't run in parallel.  I suppose running them in parallel wouldn't actually speed anything up though.

**Pull request merged**